platform/mikro-e/dev/motion-click.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2016, Imagination Technologies Limited and/or its
+ * affiliated group companies.
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. The name of the author may not be used to endorse or promote
+ *    products derived from this software without specific prior
+ *    written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS
+ * OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+ * GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * 	Motion click driver.
+ *	http://www.mikroe.com/click/motion/
+ *
+ */
+
+#ifndef __MOTION_SENSOR_H__
+#define __MOTION_SENSOR_H__
+
+/* Motion sensor is connected to pin RD0 */
+#define MOTION_SENSOR_PORT      D
+#define MOTION_SENSOR_PIN       0
+
+/* To initialize interrupt on Motion detected */
+#define MOTION_SENSOR_IRQ_INIT() __MOTION_SENSOR_IRQ_INIT()
+#define __MOTION_SENSOR_IRQ_INIT()                                      \
+  do {                                                                  \
+    CNCONDSET = _CNCOND_ON_MASK;                                        \
+    IEC1SET = _IEC1_CNDIE_MASK;                                         \
+    IFS1CLR = _IFS1_CNDIF_MASK;                                         \
+    IPC8SET = (6 << _IPC8_CNIP_POSITION) | (0 << _IPC8_CNIS_POSITION);  \
+    (void)PORTD;                                                        \
+  } while(0)
+
+/* To enable Motion sensor interupt */
+#define MOTION_SENSOR_IRQ_ENABLE() __MOTION_SENSOR_IRQ_ENABLE()
+#define __MOTION_SENSOR_IRQ_ENABLE()                                    \
+  do {                                                                  \
+    CNENDSET = _CNEND_CNIED0_MASK;                                      \
```

Comment:
should be concatenating ##MOTION_SENSOR_PORT## instead of writing D anywhere here and ##MOTION_SENSOR_PIN for 0